network-interface/src/network.rs

```diff
@@ -157,16 +158,18 @@ pub trait Network: Send + Sync + Unpin + 'static {
         T: Topic + Sync;
 
     /// Gets a value from the distributed hash table
-    async fn dht_get<K, V>(&self, k: &K) -> Result<Option<V>, Self::Error>
+    async fn dht_get<K, V, T>(&self, k: &K) -> Result<Option<V>, Self::Error>
     where
         K: AsRef<[u8]> + Send + Sync,
-        V: Deserialize + Send + Sync;
+        V: Deserialize + Send + Sync + TaggedSignable + PartialOrd,
+        T: TaggedKeyPair + Send + Sync + Serialize;
```

Comment:
This type parameter is a bit annoying because you shouldn't really change it from call to call, otherwise you'd get weird effects. I don't have a proposal how we could fix it. Do you know any alternatives?

Reply:
In principle I wanted to keep the `DHT` interface generic in terms of the data that we're putting/getting including what kind of keys you use for signing messages. So yeah, not sure how to achieve both things.